Documentation/network/lb-ipam.rst

```diff
@@ -0,0 +1,382 @@
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchExpressions:
+          - {key: color, operator: In, values: [blue, cyan]}
+    ---
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "red-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchLabels:
+          color: red
+
+There are a few special purpose selector fields which don't match on labels but
+instead on other metadata like ``.meta.name`` or ``.meta.namespace``.
+
+=============================== ===================
+Selector                        Field
+------------------------------- -------------------
+io.kubernetes.service.namespace ``.meta.namespace``
+io.kubernetes.service.name      ``.meta.name``
+=============================== ===================
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      serviceSelector:
+        matchLabels:
+          "io.kubernetes.service.namespace": "tenant-a"
+
+Conflicts
+---------
+
+IP Pools are not allowed to have overlapping CIDRs. When an administrator does
+create pools which overlap, a soft error is caused. The last added pool will be
+marked as ``Conflicting`` and no further allocation will happen from that pool.
+Therefor administrators should always check the status of all pools after making
+modifications.
+
+If we for example add 2 pools(``blue-pool`` and ``red-pool``) both with the same
+CIDR, we will see the following:
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools
+    NAME        DISABLED   CONFLICTING   IPS AVAILABLE   AGE
+    blue-pool   false      false         254             25m
+    red-pool    false      true          254             11s
+
+The reason for the conflict is stated in the status and can be accessed like so
+
+.. code-block:: shell-session
+
+    $ kubectl get ippools/red-pool -o jsonpath='{.status.conflictReason}'
+    Pool conflicts since CIDR '20.0.10.0/24' overlaps CIDR '20.0.10.0/24' from IP Pool 'blue-pool'
+
+or
+
+.. code-block:: shell-session
+
+    $ kubectl describe ippools/red-pool
+    Name:         red-pool
+    #[...]
+    Status:
+      #[...]
+      Conflict Reason:  Pool conflicts since CIDR '20.0.10.0/24' overlaps CIDR '20.0.10.0/24' from IP Pool 'blue-pool'
+      Conflicting:      true
+      #[...]
+
+Disabling a Pool
+-----------------
+
+IP Pools can be disabled. Disabling a pool will stop LB IPAM from allocating
+new IPs from the pool, but doesn't remove existing allocations. This allows
+an administrator to slowly drain pool or reserve a pool for future use.
+
+.. code-block:: yaml
+
+    apiVersion: "cilium.io/v2alpha1"
+    kind: CiliumLoadBalancerIPPool
+    metadata:
+      name: "blue-pool"
+    spec:
+      cidrs:
+      - cidr: "20.0.10.0/24"
+      disabled: true
```

Comment:
MetalLB calls this `autoAssign: false`, which I think is a clearer name.

Reply:
(This is because "disabled" could mean lots of things, and you might want to avoid assigning to a pool for other reasons than draining). See https://metallb.universe.tf/configuration/_advanced_ipaddresspool_config/

Reply:
Just following the [CFP](https://docs.google.com/document/d/1o0s2FdehujQYEGNk9znMh2L3D7VAa24bNcz60JcMuFw/edit?disco=AAAAdqpYAJ4), don't shoot the programmer :).
@ldelossa, what do you think?

Reply:
hm, I wouldn't be opposed to the naming alignment. I'm also not sure we want to use MetalLB as the canonical naming example. 

We want a term here that signifies "do not allocate anymore from this pool". I'm not getting tha from "auoAssign". That just makes me think, it won't happen automatically, but can happen manually?